libtenzir/src/type.cpp

```diff
@@ -3541,6 +3543,59 @@ record_type flatten(const record_type& type) noexcept {
   return record_type{fields};
 }
 
+auto unify(const type& a, const type& b) -> std::optional<type> {
+  // TODO: This function does not preserve metadata.
+  // TODO: Do we want to unify number types?
```

Comment:
I think yes. Otherwise it's pretty easy to write a list of numbers that doesn't work as expected.

Reply:
We agreed to do this at a later point in time.